pkg/hubble/metrics/api/api.go

```diff
@@ -64,56 +58,57 @@ type Handler interface {
 
 	// Status returns the configuration status of the metric handler
 	Status() string
-}
 
-// FlowProcessor is a metric handler which requires flows to perform metrics
-// accounting.
-// It is called upon receival of raw event data and is responsible
-// to perform metrics accounting according to the scope of the metrics plugin.
-type FlowProcessor interface {
+	// Deinit deregisters the metrics from the Prometheus registry
+	// and cleans up internal handler state
+	Deinit(registry *prometheus.Registry) bool
```

Comment:
`Deinit` should probably return an error instead of a bool

Reply:
The underlying function doesn't return an error and only returns false when trying to unregister a non-existent collector.
`func (r *Registry) Unregister(c Collector) bool {` 
Should I change it regardless?

Reply:
I was thinking because of https://github.com/cilium/cilium/pull/35185#discussion_r1815581863 we should return an error indicating what failed. Eg: if we have many metrics, we should specify which failed to unregister so it can be debugged.

Reply:
Update to return error, not bool